ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/Transaction.java

```diff
@@ -409,16 +414,23 @@ public Optional<BigInteger> getChainId() {
    * @return the transaction sender
    */
   @Override
-  public Address getSender() {
+  public synchronized Address getSender() {
```

Comment:
probably synchronized could enforce only when `sender==null`

Reply:
As I add another optimization to precalculate transations senders and hashes in parallel an asynchronously, I deleted synchronized to avoid too much overhead